AxonDeepSeg/data_management/input_data.py

```diff
@@ -1,437 +1,131 @@
-        """
-        Input:
-            trainingset_path : string or pathlib.Path: path to the trainingset
-                               folder containing 2 folders Validation and Train
-                               with images and ground truthes.
-            type_ : string 'train' or 'validation' : for the network's training.
-            thresh_indices : list of float in [0,1] : the thresholds for the ground truthes labels.
-            preload_all : if put to True, will load every image into the memory.
-        Output:
-            None.
-        """
-
-        if type_ == 'train' : # Data for train
-            self.path = Path(trainingset_path) / 'Train'
-            self.set_size = len([f for f in self.path.iterdir() if f.match('*image*')])
-            self.each_sample_once = False
-
-        if type_ == 'validation': # Data for validation
-            self.path = Path(trainingset_path) / 'Validation'
-            self.set_size = len([f for f in self.path.iterdir() if  f.match('*image*')])
-            self.each_sample_once = True
-
-        self.size_image = config["trainingset_patchsize"]
-        self.n_labels = 2
-        self.samples_seen = 0
-        self.thresh_indices = config["thresholds"]
+class DataGen(keras.utils.Sequence):
+    def __init__(self, ids, path, batch_size=8, image_size=512, thresh_indices=[0, 0.2, 0.8]):
+        self.ids = ids
+        self.path = path
         self.batch_size = batch_size
-        self.samples_list = self.reset_set(type_=type_)
-        self.epoch_size = len(self.samples_list)
-        self.preload_all = preload_all
-        self.loaded_data = None
-        #self.mean = config['dataset_mean']
-        #self.variance = config['dataset_variance']
-
-        # Loading all images if asked so
-
-        if preload_all:
-            self.loaded_data = {}
-            for id_image in self.samples_list:
-                # We are reading directly the images. Range of values : 0-255
-                image = self.read_image('image', id_image)
-                mask = self.read_image('mask', id_image)
-                self.loaded_data.update({str(id_image):[image,mask]})
-
-    def get_size(self):
-        return self.set_size
-
-    def reset_set(self, type_= 'train', shuffle=True):
-        """
-        Reset the set.
-        :param shuffle: If True, the set is shuffled, so that each batch won't systematically contain the same images.
-        :return list: List of ids of training samples
-        """
-
-        self.sample_seen = 0
+        self.image_size = image_size
+        self.on_epoch_end()
+        self.thresh_indices = thresh_indices
 
-        if type_ == 'train':
-            # Generation of a shuffled list of images
-            samples_list = list(range(self.set_size))
-            if shuffle:
-                np.random.shuffle(samples_list)
+    def __load__(self, id_name):
+        ## Path
+        image_path = os.path.join(self.path, 'image_' + id_name) + ".png"
```

Comment:
We've switched from using `os.path.join` to the more modern/pythonic (Python 3) pathlib.Path() objects. See our wiki: https://github.com/neuropoly/axondeepseg/wiki/How-to-handle-paths and the following link for examples: https://medium.com/@ageitgey/python-3-quick-tip-the-easy-way-to-deal-with-file-paths-on-windows-mac-and-linux-11a072b58d5f